transcom/mymove
# B 18669 add pro gear fields to mto shipment

## [B-18669](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a905366)

INT PR [HERE](https://github.com/transcom/mymove/pull/12120)

Please note: This has been marked as draft/do not merge until the upstream dependency [B-18840](https://github.com/transcom/mymove/pull/12260/) has been successfully merged into main.

## Summary

This works adds functionality to the internal, GHC, and Prime APIs UpdateMTOShipment functions, allowing them to update the pro gear fields created in [B-18840](https://github.com/transcom/mymove/pull/12055) and revised [here](https://github.com/transcom/mymove/pull/12111)

Is there anything you would like reviewers to give additional scrutiny?

Please ensure the updated information is returned in the 200 api response.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Ensure there is a customer with at least one move with an MTO Shipment ready to be updated in a gbloc of your choice.
2. Log in as that customer
3. In the network log in your browser's developer tools, find the MTOShipment log, and pull the id and eTag fields. NOTE: You will need to update the eTag every time you make a change to the MTOShipment.
4. Visit the internal swagger ui or use your method of choice for testing the internal api, and use this API endpoint and example to update the ActualProGearWeight and ActualSpouseProGearWeight fields: 
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/a7d6a7bf-9697-4277-b697-1a70291540b4"> 
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/474f8dca-db72-44da-a414-5b8f12117641">
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/8e1cdb53-4701-4c07-97fd-817c4786fb53">
5. Verify a 200 response and the updated fields are returned in the response.
6. Repeat steps 3, 4, and 5 for the GHC/office API and the Prime V2 API

Office examples:
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/5b0f09cd-6bc9-48dd-a152-bc8e27c65797">
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/28ba87fe-93e5-4a75-8fed-190e5100401c">
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/bd9051c9-8d8f-4f69-a621-dda4ed1f5b92">
Prime examples:
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/4ec59c6d-90e3-4615-b4ec-e1433bc2c84c">
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/a76b8638-c21e-4029-a9c7-6a6742514837">
<img width="1446" alt="image" src="https://github.com/transcom/mymove/assets/148906585/7ee4e658-4f21-435d-9182-e67d81fb1a35">


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
